src/main/java/com/descope/sdk/auth/impl/AuthenticationsBase.java

```diff
@@ -201,19 +201,20 @@ AuthenticationInfo getAuthenticationInfo(JWTResponse jwtResponse) {
   }
 
   @SuppressWarnings("unchecked")
-  boolean isTenantAssociated(Token token, String tenant) {
+  public List<String> getTenantIds(Token token) {
     if (MapUtils.isEmpty(token.getClaims())) {
-      return false;
+      return Collections.emptyList();
     }
     var claims = token.getClaims();
     if (claims.get(TENANTS_CLAIM_KEY) == null) {
-      return false;
+      return Collections.emptyList();
     }
     claims = (Map<String, Object>) claims.get(TENANTS_CLAIM_KEY);
-    if (claims.get(tenant) == null) {
-      return false;
-    }
-    return true;
+    return List.copyOf(claims.keySet());
+  }
+
+  boolean isTenantAssociated(Token token, String tenant) {
```

Comment:
What about permission checks ? 
Dont you want to add such a function ?

Reply:
We have permission checks - this is just to list the tenants themselves